android/src/main/java/com/orientationdirector/implementation/OrientationDirectorImpl.kt

```diff
@@ -126,9 +126,11 @@ class OrientationDirectorImpl internal constructor(private val context: ReactApp
   }
 
   private fun onOrientationChanged(rawDeviceOrientation: Int) {
```

Comment:
I haven't had the chance to test it out yet (waiting on an android device), but I was thinking that since the issue is related to a case where the user is rotating the device in a position that returns **UNKNOWN** (such as **flat on a surface**, but technically in portrait/landscape) wouldn't it make sense to simply **early return** when the device orientation is **UNKNOWN**?

Like so:

```kotlin
private fun onOrientationChanged(rawDeviceOrientation: Int) {
    val deviceOrientation = mUtils.convertToDeviceOrientationFrom(rawDeviceOrientation)
    if (deviceOrientation == Orientation.UNKNOWN || lastDeviceOrientation == deviceOrientation) {
      return
    }

    lastDeviceOrientation = deviceOrientation
    mEventManager.sendDeviceOrientationDidChange(deviceOrientation.ordinal)
    adaptInterfaceTo(deviceOrientation)
}
```

Doing so would:

1.  Ensure that the mUtils.convertToDeviceOrientationFrom is based solely on the rawDeviceOrientation and it actually returns a value that matches the device rotation based on its Y rotation;
2. Not override the lastDeviceOrientation with deviceOrientation's one when the latter is UNKNOWN;
3. Skip an update when the deviceOrientation matches the lastDeviceOrientation's one, like you coded in the if check;

What do you think? 
Does it make sense for you?

Reply:
> I haven't had the chance to test it out yet (waiting on an android device)

I've managed to reproduce it in the example app and I can confirm that your fix does work, as well as the other implementation that I suggested.

I'll wait to hear from you about what you think of it and then we can get this merged and release a new version.

Reply:
I think if the orientation of the device returned from `onOrientationChanged` is actually `ORIENTATION_UNKNOWN`, it should override the lastDeviceOrientation.

How about this:

```kotlin
private fun onOrientationChanged(rawDeviceOrientation: Int) {
    val deviceOrientation = mUtils.getDeviceOrientationFrom(rawDeviceOrientation)
    if (rawDeviceOrientation != -1 && deviceOrientation == Orientation.UNKNOWN || lastDeviceOrientation == deviceOrientation) {
        return
    }
    lastDeviceOrientation = deviceOrientation
    mEventEmitter.sendDeviceOrientationDidChange(deviceOrientation.ordinal)
    adaptInterfaceTo(deviceOrientation)
  }
```

Reply:
Good catch!
It does make sense to actually let it print unknown when it can't be determined.
Per documentation it should print it when the device is close to a flat position.

I would leverage the _ORIENTATION_UNKNOWN_ constant imported from: 
```kotlin
import android.view.OrientationEventListener.ORIENTATION_UNKNOWN
```

And I'd update the code as follows to enhance readability:

```kotlin
private fun onOrientationChanged(rawDeviceOrientation: Int) {
    val deviceOrientation = mUtils.convertToDeviceOrientationFrom(rawDeviceOrientation)

    if (rawDeviceOrientation != ORIENTATION_UNKNOWN && deviceOrientation == Orientation.UNKNOWN) {
      return;
    }

    if (lastDeviceOrientation == deviceOrientation) {
      return
    }

    lastDeviceOrientation = deviceOrientation
    mEventManager.sendDeviceOrientationDidChange(deviceOrientation.ordinal)
    adaptInterfaceTo(deviceOrientation)
}
```

What do you think?
If you can make the changes and test it out on a real device, we can get this merged.

Again, thank you!

Reply:
It looks perfect!
I updated the code and tested it on the real device, as expected.

Reply:
> It looks perfect! I updated the code and tested it on the real device, as expected.

I've noticed that you deleted the following instructions:

```kotlin
mEventManager.sendDeviceOrientationDidChange(deviceOrientation.ordinal)
lastDeviceOrientation = deviceOrientation
```

Don't we still need them to properly notify js side? Perhaps it was a typo?

Reply:
Sorry, that's a mistake😂